1. 1. Scenario: A company wants to analyze the sales performance of its products in different regions. They have collected the following data:
   
   Region A: [10, 15, 12, 8, 14]
   
   Region B: [18, 20, 16, 22, 25]
   
   Calculate the mean sales for each region.


Region A:

Sales data: [10, 15, 12, 8, 14]

Total sales in Region A = 10 + 15 + 12 + 8 + 14 = 59

Number of data points in Region A = 5

Mean sales in Region A = Total sales in Region A / Number of data points in Region A = 59 / 5 = 11.8

Region B:

Sales data: [18, 20, 16, 22, 25]

Total sales in Region B = 18 + 20 + 16 + 22 + 25 = 101

Number of data points in Region B = 5

Mean sales in Region B = Total sales in Region B / Number of data points in Region B = 101 / 5 = 20.2

2. Scenario: A survey is conducted to measure customer satisfaction on a scale of 1 to 5. The data collected is as follows:

   [4, 5, 2, 3, 5, 4, 3, 2, 4, 5]
   |
   Calculate the mode of the survey responses.


In this case, both 4 and 5 appear most frequently, with a count of three each. Since there is more than one mode, this data set is considered multimodal. Therefore, the modes of the survey responses are 4 and 5.

3. Scenario: A company wants to compare the salaries of two departments. The salary data for Department A and Department B are as follows:
    
   Department A: [5000, 6000, 5500, 7000]

   Department B: [4500, 5500, 5800, 6000, 5200]
    
   Calculate the median salary for each department.



Department A:
    
    
Salary data: [5000, 6000, 5500, 7000]

Arrange in ascending order: [5000, 5500, 6000, 7000]

Since the number of data points is even, the median is the average of the two middle values.

Median salary in Department A = (5500 + 6000) / 2 = 5750

Department B:
    
Salary data: [4500, 5500, 5800, 6000, 5200]

Arrange in ascending order: [4500, 5200, 5500, 5800, 6000]

Since the number of data points is odd, the median is the middle value.

Median salary in Department B = 5500

4. Scenario: A data analyst wants to determine the variability in the daily stock prices of a company. The data collected is as follows:

   [25.5, 24.8, 26.1, 25.3, 24.9]
   
   Calculate the range of the stock prices.


Range = Highest value - Lowest value

Range = 26.1 - 24.8

Range = 1.3

5. Scenario: A study is conducted to compare the performance of two different teaching methods. The test scores of the students in each group are as follows:

   Group A: [85, 90, 92, 88, 91]
   
   Group B: [82, 88, 90, 86, 87]
   
   Perform a t-test to determine if there is a significant difference in the mean scores between the two groups.


In [1]:
from scipy import stats

# Data for Group A and Group B
group_A = [85, 90, 92, 88, 91]
group_B = [82, 88, 90, 86, 87]

# Perform two-sample t-test
t_statistic, p_value = stats.ttest_ind(group_A, group_B)

# Print the results
print("T-Statistic:", t_statistic)
print("P-Value:", p_value)


T-Statistic: 1.4312528946642733
P-Value: 0.19023970239078333


Since the p-value (0.19023970239078333) is greater than the commonly used significance level of 0.05, we do not have enough evidence to reject the null hypothesis. Therefore, we cannot conclude that there is a significant difference in the mean scores between Group A and Group B based on this data.

6. Scenario: A company wants to analyze the relationship between advertising expenditure and sales. The data collected is as follows:
    
   Advertising Expenditure (in thousands): [10, 15, 12, 8, 14]

   Sales (in thousands): [25, 30, 28, 20, 26]
    
   Calculate the correlation coefficient between advertising expenditure and sales.


In [2]:
import numpy as np

# Advertising Expenditure (in thousands)
advertising_expenditure = [10, 15, 12, 8, 14]

# Sales (in thousands)
sales = [25, 30, 28, 20, 26]

# Calculate the correlation coefficient
correlation_coefficient = np.corrcoef(advertising_expenditure, sales)[0, 1]

# Print the result
print("Correlation Coefficient:", correlation_coefficient)


Correlation Coefficient: 0.8757511375750132


7. Scenario: A survey is conducted to measure the heights of a group of people. The data collected is as follows:
   [160, 170, 165, 155, 175, 180, 170]
   Calculate the standard deviation of the heights.


Calculate the mean:

Mean = (160 + 170 + 165 + 155 + 175 + 180 + 170) / 7 = 166.43 (rounded to two decimal places)


Subtract the mean and square the differences:

(160 - 166.43)^2 = 42.1749


(170 - 166.43)^2 = 12.9569

(165 - 166.43)^2 = 1.9449

(155 - 166.43)^2 = 130.5169

(175 - 166.43)^2 = 74.0769

(180 - 166.43)^2 = 185.1169

(170 - 166.43)^2 = 12.9569

Calculate the mean of the squared differences:

Mean of squared differences = (42.1749 + 12.9569 + 1.9449 + 130.5169 + 74.0769 + 185.1169 + 12.9569) / 7 = 61.6735 (rounded to four decimal places)

Take the square root of the mean of squared differences:

Standard deviation = sqrt(61.6735) = 7.8533

8. Scenario: A company wants to analyze the relationship between employee tenure and job satisfaction. The data collected is as follows:
    
   Employee Tenure (in years): [2, 3, 5, 4, 6, 2, 4]

   Job Satisfaction (on a scale of 1 to 10): [7, 8, 6, 9, 5, 7, 6]
    
   Perform a linear regression analysis to predict job satisfaction based on employee tenure.



In [3]:
import statsmodels.api as sm
import numpy as np

# Employee Tenure (in years)
employee_tenure = [2, 3, 5, 4, 6, 2, 4]

# Job Satisfaction (on a scale of 1 to 10)
job_satisfaction = [7, 8, 6, 9, 5, 7, 6]

# Add a constant term to the predictor variable
employee_tenure = sm.add_constant(employee_tenure)

# Fit the linear regression model
model = sm.OLS(job_satisfaction, employee_tenure)
results = model.fit()

# Get the predicted job satisfaction values
predicted_job_satisfaction = results.predict(employee_tenure)

# Print the regression results
print(results.summary())


                            OLS Regression Results                            
Dep. Variable:                      y   R-squared:                       0.271
Model:                            OLS   Adj. R-squared:                  0.125
Method:                 Least Squares   F-statistic:                     1.859
Date:                Tue, 11 Jul 2023   Prob (F-statistic):              0.231
Time:                        08:05:18   Log-Likelihood:                -10.362
No. Observations:                   7   AIC:                             24.72
Df Residuals:                       5   BIC:                             24.62
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const          8.5957      1.361      6.316      0.0

/opt/conda/lib/python3.10/site-packages/statsmodels/stats/stattools.py:74: ValueWarning: omni_normtest is not valid with less than 8 observations; 7 samples were given.
  warn("omni_normtest is not valid with less than 8 observations; %i "


9. Scenario: A study is conducted to compare the effectiveness of two different medications. The recovery times of the patients in each group are as follows:

   Medication A: [10, 12, 14, 11, 13]
   
   Medication B: [15, 17, 16, 14, 18]
   
   Perform an analysis of variance (ANOVA) to determine if there is a significant difference in the mean recovery times between the two medications.


In [4]:
from scipy import stats

# Recovery times for Medication A and Medication B
medication_A = [10, 12, 14, 11, 13]
medication_B = [15, 17, 16, 14, 18]

# Perform one-way ANOVA
f_value, p_value = stats.f_oneway(medication_A, medication_B)

# Print the results
print("F-value:", f_value)
print("P-value:", p_value)


F-value: 16.0
P-value: 0.003949772803445326


10. Scenario: A company wants to analyze customer feedback ratings on a scale of 1 to 10. The data collected is

 as follows:
 
    [8, 9, 7, 6, 8, 10, 9, 8, 7, 8]
    
    Calculate the 75th percentile of the feedback ratings.


In [1]:
import numpy as np

# Feedback ratings
feedback_ratings = [8, 9, 7, 6, 8, 10, 9, 8, 7, 8]

# Calculate the 75th percentile
percentile_75 = np.percentile(feedback_ratings, 75)

# Print the result
print("75th percentile:", percentile_75)


75th percentile: 8.75


11.  Scenario: A quality control department wants to test the weight consistency of a product. The weights of a sample of products are as follows:
    
    [10.2, 9.8, 10.0, 10.5, 10.3, 10.1]
    
    Perform a hypothesis test to determine if the mean weight differs significantly from 10 grams.


In [2]:
from scipy import stats

# Weights of the sample products
weights = [10.2, 9.8, 10.0, 10.5, 10.3, 10.1]

# Perform one-sample t-test
t_statistic, p_value = stats.ttest_1samp(weights, 10)

# Print the results
print("T-Statistic:", t_statistic)
print("P-Value:", p_value)


T-Statistic: 1.5126584522688367
P-Value: 0.19077595151110102


12. Scenario: A company wants to analyze the click-through rates of two different website designs. The number of clicks for each design is as follows:

    Design A: [100, 120, 110, 90, 95]
    
    Design B: [80, 85, 90, 95, 100]
    
    Perform a chi-square test to determine if there is a significant difference in the click-through rates between the two designs.


In [5]:
from scipy import stats
import numpy as np

# Click-through rates for Design A and Design B
design_A = [100, 120, 110, 90, 95]
design_B = [80, 85, 90, 95, 100]

# Combine the data into a contingency table
observed = np.array([design_A, design_B])

# Perform chi-square test
chi2_statistic, p_value = stats.chisquare(observed)

# Print the results
print("Chi-square statistic:", chi2_statistic)
print("P-value:", p_value)



Chi-square statistic: [2.22222222 5.97560976 2.         0.13513514 0.12820513]
P-value: [0.13603713 0.01450507 0.15729921 0.71316606 0.72030033]


13. scenario: A survey is conducted to measure customer satisfaction with a product on a scale of 1 to 10. The data collected is as follows:
    
    [7, 9, 6, 8, 10, 7, 8, 9, 7, 8]
    
    Calculate the 95% confidence interval for the population mean satisfaction score.


To calculate the 95% confidence interval for the population mean satisfaction score, we can use the following formula:

Confidence Interval = x̄ ± Z * (σ/√n)

Where:

x̄ is the sample mean
Z is the z-score corresponding to the desired confidence level (95% confidence level corresponds to a z-score of 1.96)
σ is the standard deviation of the population (unknown in this case)
n is the sample size
Let's calculate it step by step:

Calculate the sample mean (x̄):
x̄ = (7 + 9 + 6 + 8 + 10 + 7 + 8 + 9 + 7 + 8) / 10 = 79 / 10 = 7.9

Calculate the sample standard deviation (s):
s = √[(Σ(xi - x̄)^2) / (n - 1)]
= √[((7-7.9)^2 + (9-7.9)^2 + (6-7.9)^2 + (8-7.9)^2 + (10-7.9)^2 + (7-7.9)^2 + (8-7.9)^2 + (9-7.9)^2 + (7-7.9)^2 + (8-7.9)^2) / (10-1)]
= √[(0.81 + 1.21 + 2.89 + 0.01 + 4.41 + 0.81 + 0.01 + 1.21 + 0.81 + 0.01) / 9]
= √(12.07 / 9)
≈ √1.34
≈ 1.16

Calculate the standard error (SE):
SE = s / √n
= 1.16 / √10
≈ 0.366

Calculate the margin of error (ME):
ME = Z * SE
= 1.96 * 0.366
≈ 0.717

Calculate the lower and upper bounds of the confidence interval:
Lower bound = x̄ - ME
= 7.9 - 0.717
≈ 7.183

Upper bound = x̄ + ME
= 7.9 + 0.717
≈ 8.617

Therefore, the 95% confidence interval for the population mean satisfaction score is approximately 7.183 to 8.617.

14. Scenario: A company wants to analyze the effect of temperature on product performance. The data collected is as follows:
    
    Temperature (in degrees Celsius): [20, 22, 23, 19, 21]
    
    Performance (on a scale of 1 to 10): [8, 7, 9, 6, 8]
    
    Perform a simple linear regression to predict performance based on temperature.


In [6]:
import numpy as np
from sklearn.linear_model import LinearRegression

# Temperature data
temperature = np.array([20, 22, 23, 19, 21]).reshape(-1, 1)

# Performance data
performance = np.array([8, 7, 9, 6, 8])

# Create a linear regression object
regression_model = LinearRegression()

# Fit the regression model to the data
regression_model.fit(temperature, performance)

# Get the slope (m) and y-intercept (b)
slope = regression_model.coef_[0]
intercept = regression_model.intercept_

# Print the equation of the regression line
print(f"The equation of the regression line is: y = {slope}x + {intercept}")


The equation of the regression line is: y = 0.4999999999999999x + -2.8999999999999986


15. Scenario: A study is conducted to compare the preferences of two groups of participants. The preferences are measured on a Likert scale from 1 to 5. The data collected is as follows:

    Group A: [4, 3, 5, 2, 4]
    
    Group B: [3, 2, 4, 3, 3]
    
    Perform a Mann-Whitney U test to determine if there is a significant difference in the median preferences between the two groups.


In [7]:
import scipy.stats as stats

# Preferences data
group_a = [4, 3, 5, 2, 4]
group_b = [3, 2, 4, 3, 3]

# Perform Mann-Whitney U test
statistic, p_value = stats.mannwhitneyu(group_a, group_b, alternative='two-sided')

# Print the test results
print("Mann-Whitney U test results:")
print(f"U statistic: {statistic}")
print(f"p-value: {p_value}")


Mann-Whitney U test results:
U statistic: 17.0
p-value: 0.380836480306712


16. Scenario: A company wants to analyze the distribution of customer ages. The data collected is as follows:

    [25, 30, 35, 40, 45, 50, 55, 60, 65, 70]
    
    Calculate the interquartile range (IQR) of the ages.


Sort the data in ascending order:

[25, 30, 35, 40, 45, 50, 55, 60, 65, 70]

Find the first quartile (Q1):
Q1 is the median of the lower half of the data. Since we have an even number of data points, we take the average of the two middle values.

Q1 = (35 + 40) / 2 = 75 / 2 = 37.5

Find the third quartile (Q3):

Q3 is the median of the upper half of the data.

Q3 = (55 + 60) / 2 = 115 / 2 = 57.5

Calculate the interquartile range (IQR):

IQR = Q3 - Q1 = 57.5 - 37.5 = 20

17. Scenario: A study is conducted to compare the performance of three different machine learning algorithms. The accuracy scores for each algorithm are as follows:
    
    Algorithm A: [0.85, 0.80, 0.82, 0.87, 0.83]
    
    Algorithm B: [0.78, 0.82, 0.84, 0.80, 0.79]
    
    Algorithm C: [0.90, 0.88, 0.89, 0.86, 0.87]
    
    Perform a Kruskal-Wallis test to determine if there is a significant difference in the median accuracy scores between the algorithms.


In [8]:
import scipy.stats as stats

# Accuracy scores data
algorithm_a = [0.85, 0.80, 0.82, 0.87, 0.83]
algorithm_b = [0.78, 0.82, 0.84, 0.80, 0.79]
algorithm_c = [0.90, 0.88, 0.89, 0.86, 0.87]

# Perform Kruskal-Wallis test
statistic, p_value = stats.kruskal(algorithm_a, algorithm_b, algorithm_c)

# Print the test results
print("Kruskal-Wallis test results:")
print(f"Test statistic: {statistic}")
print(f"p-value: {p_value}")


Kruskal-Wallis test results:
Test statistic: 9.696947935368053
p-value: 0.007840333026249539


18. Scenario: A company wants to analyze the effect of price on sales. The data collected is as follows:
    
    Price (in dollars): [10, 15, 12, 8, 14]
    
    Sales: [100, 80, 90, 110, 95]
    
    Perform a simple linear regression to predict

 sales based on price.


In [11]:
import numpy as np
from sklearn.linear_model import LinearRegression

# Price data
price = np.array([10, 15, 12, 8, 14]).reshape(-1, 1)

# Sales data
sales = np.array([100, 80, 90, 110, 95])

# Create a linear regression object
regression_model = LinearRegression()

# Fit the regression model to the data
regression_model.fit(price, sales)

# Get the slope (m) and y-intercept (b)
slope = regression_model.coef_[0]
intercept = regression_model.intercept_

# Print the equation of the regression line
print(f"The equation of the regression line is: y = {slope}x + {intercept}")


The equation of the regression line is: y = -3.5060975609756095x + 136.3719512195122


19. Scenario: A survey is conducted to measure the satisfaction levels of customers with a new product. The data collected is as follows:
    
    [7, 8, 9, 6, 8, 7, 9, 7, 8, 7]
    
    Calculate the standard error of the mean satisfaction score.


In [12]:
import numpy as np

# Data
data = np.array([7, 8, 9, 6, 8, 7, 9, 7, 8, 7])

# Calculate the standard error of the mean
standard_error = np.std(data, ddof=1) / np.sqrt(len(data))

# Print the standard error of the mean
print(f"The standard error of the mean is: {standard_error}")


The standard error of the mean is: 0.30550504633038933


20. Scenario: A company wants to analyze the relationship between advertising expenditure and sales. The data collected is as follows:
    
    Advertising Expenditure (in thousands): [10, 15, 12, 8, 14]
    
    Sales (in thousands): [25, 30, 28, 20, 26]
    
    Perform a multiple regression analysis to predict sales based on advertising expenditure.



In [13]:
import numpy as np
import statsmodels.api as sm

# Advertising expenditure data
expenditure = np.array([10, 15, 12, 8, 14])

# Sales data
sales = np.array([25, 30, 28, 20, 26])

# Add a column of ones as the constant term to the independent variable
X = sm.add_constant(expenditure)

# Create a multiple regression model
model = sm.OLS(sales, X)

# Fit the model to the data
results = model.fit()

# Print the regression results
print(results.summary())


                            OLS Regression Results                            
Dep. Variable:                      y   R-squared:                       0.767
Model:                            OLS   Adj. R-squared:                  0.689
Method:                 Least Squares   F-statistic:                     9.872
Date:                Tue, 11 Jul 2023   Prob (F-statistic):             0.0516
Time:                        09:50:57   Log-Likelihood:                -9.5288
No. Observations:                   5   AIC:                             23.06
Df Residuals:                       3   BIC:                             22.28
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const         12.2012      4.429      2.755      0.0

/opt/conda/lib/python3.10/site-packages/statsmodels/stats/stattools.py:74: ValueWarning: omni_normtest is not valid with less than 8 observations; 5 samples were given.
  warn("omni_normtest is not valid with less than 8 observations; %i "
